src/Home/Crypto/ChaCha20/ChaCha20.ZC

```diff
@@ -0,0 +1,181 @@
+
+// Rotating bits left
+U32 RotL32(U32 x, I32 n) {
+	return (x << n) | (x >> (32 - n));
+}
+
+// Packing bytes into a 32-bit unsigned integer
+U32 Pack4(U8 *a) {
+    U32 res = 0;
+    res |= a[0](U32) << 0 * 8;
+    res |= a[1](U32) << 1 * 8;
+    res |= a[2](U32) << 2 * 8;
+    res |= a[3](U32) << 3 * 8;
```

Comment:
Does promotion to U32 occur after one byte at a[3] is accessed? Considering << is multiplying by a power of 2, why do you need to multiply again by 8?